File: frontend/src/types/alert.ts
export interface AlertItem {
  id: number;
  timestamp: string;

  tenant: string;
  rule_name: string;

  severity: number;

  src_ip: string;
  event_type: string;

  message: string;

  created_at: string;
}

export interface AlertResponse {
  count: number;
  alerts: AlertItem[];
}

File: backend/entity/alert.go
package entity

import "time"

type Alert struct {
	ID uint `gorm:"primaryKey" json:"id"`

	Timestamp time.Time `json:"timestamp"`

	Tenant   string `json:"tenant"`
	RuleName string `json:"rule_name"`

	Severity int `json:"severity"`

	SrcIP     string `json:"src_ip"`
	EventType string `json:"event_type"`

	Message string `json:"message"`

	CreatedAt time.Time `json:"created_at"`
}
